src/pages/resultsView/expression/expressionHelpers.ts

```diff
@@ -0,0 +1,212 @@
+    "exon14skip": "other",
+    "frame_shift_del": "frameshift",
+    "frame_shift_ins": "frameshift",
+    "frameshift": "frameshift",
+    "frameshift deletion": "frameshift",
+    "frameshift insertion": "frameshift",
+    "frameshift_coding": "frameshift",
+    "frameshift_insertion": "frameshift",
+    "fusion": "other",
+    "in_frame_del": "in_frame",
+    "in_frame_ins": "in_frame",
+    "missense": "missense",
+    "missense_mutation": "missense",
+    "nonsense": "nonsense",
+    "nonsense_mutation": "nonsense",
+    "nonstop_mutation": "nonstop",
+    "splice": "splice",
+    "splice_site": "splice",
+    "splice_site_snp": "splice",
+    "splicing": "splice",
+    "translation_start_site": "nonstart",
+    "viii deletion": "other"
+}
+
+
+export function isTCGAPubStudy(studyId:string){
+    return /tcga_pub$/.test(studyId);
+}
+
+export function isTCGAProvStudy(studyId:string){
+    return /tcga$/.test(studyId);
+}
+
+export function isPanCanStudy(studyId:string){
+    return /tcga_pan_can_atlas/.test(studyId);
+}
+
+export function getMolecularDataBuckets(studyData: NumericGeneMolecularData[],
+                                 showMutations: boolean,
+                                 mutationsKeyedBySampleId: { [sampleId: string]: Mutation },
+                                 coverageInformation:CoverageInformation,
+                                        hugoGeneSymbol:string
+                                        ) {
+
+    // if mutation mode is on, we want to fill buckets by mutation type (or unmutated)
+    const mutationModeInteratee = (memo: MolecularDataBuckets, molecularData: NumericGeneMolecularData) => {
+
+        const mutation = mutationsKeyedBySampleId[molecularData.uniqueSampleKey];
+        if (mutation) {
+            const canonicalMutationType = getCanonicalMutationType(mutation.mutationType) || "other";
+            const bucket = memo.mutationBuckets[canonicalMutationType] = memo.mutationBuckets[canonicalMutationType] || [];
+            bucket.push(molecularData);
+        } else if (isSampleProfiled(molecularData.uniqueSampleKey,molecularData.molecularProfileId,
+                hugoGeneSymbol, coverageInformation)) {
+
+            memo.unmutatedBucket.push(molecularData);
+        } else {
+            memo.unsequencedBucket.push(molecularData);
+        }
+        return memo;
+    };
+
+    // if mutation mode is off, we don't care about mutation state
+    const noMutationModeIteratee = (memo: MolecularDataBuckets, molecularData: NumericGeneMolecularData) => {
+        memo.unmutatedBucket.push(molecularData);
+        return memo;
+    };
+
+    const iteratee = (showMutations) ? mutationModeInteratee : noMutationModeIteratee;
+
+    // populate buckets using iteratee
+    const buckets = studyData.reduce(iteratee, {mutationBuckets: {}, unmutatedBucket: [], unsequencedBucket:[] });
+
+    return buckets;
+
+}
+
+export function calculateJitter(boxWidth: number){
+    return (Math.random()-0.5)*0.45
```

Comment:
see my comment in slack https://cbioportal.slack.com/archives/C359NNB41/p1528828600000715